sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -3835,6 +3835,24 @@ class SparkConnectPlanner(
         UnresolvedScalarSubqueryPlanId(planId)
       case proto.SubqueryExpression.SubqueryType.SUBQUERY_TYPE_EXISTS =>
         UnresolvedExistsPlanId(planId)
+      case proto.SubqueryExpression.SubqueryType.SUBQUERY_TYPE_TABLE_ARG =>
+        if (!getSubqueryExpression.hasTableArgOptions) {
+          throw InvalidPlanInput("Missing tableArgOptions for SUBQUERY_TYPE_TABLE_ARG")
+        }
```

Comment:
Why is this invalid? Isn't `TableArgOptions` optional?

Reply:
Ah I see what you meant, adjusted.